### PR TITLE
Changing demo.md from keywords to runes

### PR DESCRIPTION
**Pull request by joshuareagan**

### CODE REVIEW

**Comment by Fang-** [comment]
Good stuff! My only little gripe with this is that you introduce the term `rune` and a link to the relevant documentation on line four, instead of right off the bat. Not sure if you want to integrate this into the introductory paragraph or just say "`|= [rune](#)`".

**Comment by joshuareagan** [comment]
yeah that's definitely confusing...

**Comment by joshuareagan** [comment]
Fixed

File: docs/hoon/demo.md [hunk 1]
```diff
@@ -41,33 +38,38 @@ The code produces a list in which each item is either a number
...
+*[rune](https://urbit.org/docs/about/glossary#rune)* to define a 
+function of one argument, `end`, which is of the type `atom` (i.e. 
+an unsigned integer).
 
 Line 2 declares a variable, `count`, with initial value `1`.
 
 Line 3 begins a loop.
 
 Line 4 defines the product of the loop as a list of `tape`s. 
 (A `tape` is a list of characters, one of Hoon's two string
```
Comment (joemfb) [inline docs/hoon/demo.md]: This is not technically correct; `(list @c)` is a list of "characters". A type is a list of character-bytes, which can be a very different thing. Probably not relevant here, but I just can't help myself ...

Comment (joshuareagan) [reply docs/hoon/demo.md]: There's a lot that isn't technically correct in this doc.  Hoon doesn't have loops!

Comment (Fang-) [reply docs/hoon/demo.md]: I think it's fair to trade in a bit of technical correctness for some slightly slimmer semantics.

**Pull request merged**